File: backend/ClickStitch/Models/Mappers/ThreadMapper.cs
using Data.Records;

namespace ClickStitch.Models.Mappers;

public static class ThreadMapper
{
    public static ThreadModel Map(ThreadRecord thread) => new()
    {
        Reference = thread.Reference,
        Brand = thread.Brand,
        Code = thread.Code,
        Colour = thread.Colour
    };
}

File: frontend/src/api/mappers/Thread.mapper.ts
import type { IThread } from '@/models/Thread.model';
import type { IApiThread } from '@/api/api-models/ApiThread.type';

export const threadMapper = {

    map(thread: IApiThread): IThread {
        return {
            reference: thread.reference,
            brand: thread.brand,
            code: thread.code,
            colour: thread.colour,
        };
    },

};